app/models/review.rb

```diff
@@ -28,4 +28,22 @@ def sum_of_votes
     end
     sum
   end
+
+  def user_upvoted?(user)
```

Comment:
Also just as a stylistic thing I tend to only use `unless` if I'm doing an inline operation like:
```
return unless flowdock.open?
```

if you are nesting conditional logic like below I'd use the following syntax:
```
if !(user.nil?)
    vote = votes.find_by(user_id: user.id)
    if !(vote.nil?)
        vote.value == 1
    end
end
```